src/osd/ECBackend.cc

```diff
@@ -996,15 +1005,36 @@ void ECBackend::handle_sub_read(
 	goto error;
       }
     }
+
     for (auto j = i->second.begin(); j != i->second.end(); ++j) {
       bufferlist bl;
-      r = store->read(
-	ch,
-	ghobject_t(i->first, ghobject_t::NO_GEN, shard),
-	j->get<0>(),
-	j->get<1>(),
-	bl, j->get<2>(),
-	true); // Allow EIO return
+      if((op.subchunks.find(i->first)->second.size() == 1) && (op.subchunks.find(i->first)->second.front().second == ec_impl->get_sub_chunk_count())) {
```

Comment:
please clean up the whitespace here (space after if, 80 chars per line), etc.

also use the fancy for loop syntax 'for (auto myvar : thecontainer)' where it helps readability